openlibrary/plugins/upstream/models.py

```diff
@@ -413,14 +413,19 @@ def wp_citation_fields(self):
         result = {
             "title": self.works[0].title.replace("[", "&#91").replace("]", "&#93"),
             "publication-date": self.get('publish_date'),
-            "url": "http://openlibrary.org%s" % self.url()
+            "ol": str(self.get_olid()),
         }
+        data = self.get_ia_meta_fields()
+
+
+        if data['identifier-access']:
+            result['url'] = data['identifier-access']
```

Comment:
I realise this URL is taken from archive.org metadata, but it should be https://archive.org/details/<ocaid>  This metadata field won't be populated in most(?) records, and looks like it is out of date where it exists. Generating the current URL from the data we know will be better.

the archive.org id <ociad> should be available at`self.ocaid`